nncf/onnx/graph/nncf_graph_builder.py

```diff
@@ -74,6 +78,9 @@ def has_bias(self) -> bool:
     def has_node_attrs(self) -> bool:
         return bool(self.node_attrs)
 
+    def get_backend_agnostic_attributes(self) -> BaseLayerAttributes:
+        return self._layer_attributes
```

Comment:
Again, the `ONNXLayerAttributes` is already a `BaseLayerAttribute` and according to L in [SOLID](https://en.wikipedia.org/wiki/SOLID) should be usable as the latter everywhere; yet this function seemingly returns some other thing instead. 

I expect that at the stage where common code executes, and where it has to look into layer attributes, this common code only knows about the "base" portion of the layer attributes, which is already by definition backend-agnostic. If there are any kinds of `isinstance` checks for backend-specific versions of `LayerAttributes`, then this was most likely introduced as a crutch and should be removed, not proliferated.

Reply:
Please provide an explicit example on how this should work in my case. Layer attributes unification for all backend was discussed before, the decision was to put this refactoring task to refactoring list as all PTQ algorithms are potentially affected by the refactoring

Reply:
This is explicit:
> I expect that at the stage where common code executes, and where it has to look into layer attributes, this common code only knows about the "base" portion of the layer attributes, which is already by definition backend-agnostic. If there are any kinds of isinstance checks for backend-specific versions of LayerAttributes, then this was most likely introduced as a crutch and should be removed, not proliferated.

but I'll restate this for you. Common code entities should be backend-agnostic already. There should be no methods such a `get_backend_agnostic_anything` in common structures and classes. If you are going to introduce a `class ONNXLayerAttributes(BaseLayerAttributes)`, then make sure it satisfies L in SOLID, e.g. that it can be used as-is, without additional method calls, when the code utilizing an instance of `ONNXLayerAttributes` needs only the `BaseLayerAttributes` part of this object.

> the decision was to put this refactoring task to the refactoring

Ok, if I understood correctly you do not want to or are unable to do the refactoring in anything related to this. Don't touch this at all, then, because, evidently, since you need to touch `BaseLayerAttributes` here, your changes affect all PTQ algos too. I've given multiple ideas on how to do constant filtering without impacting layer attributes.

Reply:
Outdated